From 7494917a04f2a5fad42f68b5ea2898fa189b3e1a Mon Sep 17 00:00:00 2001
From: Greg Rose <gregory.v.rose@intel.com>
Date: Sat, 19 Jan 2013 06:40:22 +0000
Subject: [PATCH 699/858] ixgbevf: Fix link speed message to support 100Mbps

The X540 can link at 100Mbps - fix the link speed indicator message to
show that value.

Signed-off-by: Greg Rose <gregory.v.rose@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit b876a744a228308ecf75466831453f0863d253db)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c b/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
index 551e31d..c3db6cd 100644
--- a/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
+++ b/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
@@ -2245,10 +2245,23 @@ static void ixgbevf_watchdog_task(struct work_struct *work)
 
  if (link_up) {
   if (!netif_carrier_ok(netdev)) {
+   char *link_speed_string;
+   switch (link_speed) {
+   case IXGBE_LINK_SPEED_10GB_FULL:
+    link_speed_string = "10 Gbps";
+    break;
+   case IXGBE_LINK_SPEED_1GB_FULL:
+    link_speed_string = "1 Gbps";
+    break;
+   case IXGBE_LINK_SPEED_100_FULL:
+    link_speed_string = "100 Mbps";
+    break;
+   default:
+    link_speed_string = "unknown speed";
+    break;
+   }
    dev_info(&adapter->pdev->dev,
-    "NIC Link is Up, %u Gbps\n",
-    (link_speed == IXGBE_LINK_SPEED_10GB_FULL) ?
-    10 : 1);
+    "NIC Link is Up, %s\n", link_speed_string);
    netif_carrier_on(netdev);
    netif_tx_wake_all_queues(netdev);
   }
-- 
1.8.3.1